examples/feature_selection/plot_permutation_test_for_classification.py

```diff
@@ -22,48 +20,115 @@
 from sklearn.svm import SVC
 from sklearn.model_selection import StratifiedKFold
 from sklearn.model_selection import permutation_test_score
-from sklearn import datasets
+from sklearn.datasets import load_iris
+
 
+# %%
+# Dataset
+# -------
+#
+# We will use the :ref:`iris_dataset`, which consists of measurements taken
+# from 3 types of irises.
 
-# #############################################################################
-# Loading a dataset
-iris = datasets.load_iris()
+iris = load_iris()
 X = iris.data
 y = iris.target
 n_classes = np.unique(y).size
 
-# Some noisy data not correlated
-random = np.random.RandomState(seed=0)
-E = random.normal(size=(len(X), 2200))
+# %%
+# We will also generate some random feature data, uncorrelated with the class
+# labels in the iris dataset.
+
+rng = np.random.RandomState(seed=0)
+X_rand = rng.normal(size=(len(X), 2200))
 
-# Add noisy data to the informative features for make the task harder
-X = np.c_[X, E]
+# %%
+# Permutation test score
+# ----------------------
+#
+# Next, we calculate the
+# :func:`~sklearn.model_selection.permutation_test_score` using the original
+# iris dataset, which strongly predict the labels and
+# the randomly generated features and iris labels, which should have
+# no dependency between features and labels. The
+# :class:`~sklearn.svm.SVC` classifier and :ref:`accuracy_score` are used.
+#
+# :func:`~sklearn.model_selection.permutation_test_score` generates a null
+# distribution by calculating the accuracy of the classifier
+# on 1000 different permutations of the dataset, where features
+# remain the same but labels undergo different permutations. This is the
+# distribution for the null hypothesis which states there is no dependency
+# between the features and labels. An empirical p value is then calculated as
+# the percentage of permutations for which the score obtained is greater
+# that the score obtained using the original data.
 
-svm = SVC(kernel='linear')
+clf = SVC(kernel='linear', random_state=7)
 cv = StratifiedKFold(2)
```

Comment:
It might be best to add the `shuffle=True` with `random_state=0` in the cv. `iris` is not shuffled by default and I don't like to use it in example without shuffling it :)